File: Blind_75_List/Intervals/435. Non-overlapping Intervals.py
# Problem Link: https://leetcode.com/problems/non-overlapping-intervals/

class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        ans = 0
        prev_max = -float("inf")
        intervals.sort()
        for [a,b] in intervals:
            if a >= prev_max:
                prev_max = b
            else:
                ans+=1
                prev_max = min(prev_max, b)
        return ans

File: Blind_75_List/Intervals/435. Non-overlapping Intervals.cpp
//Problem Link: https://leetcode.com/problems/non-overlapping-intervals/

class Solution {
public:
    const int inf = 1e9 + 3;
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int prev_max = -inf;
        int ans = 0;
        for(auto& p: intervals){
            if (p[0] >= prev_max){
                prev_max = p[1];
            }
            else{
                ans++;
                prev_max = min(prev_max, p[1]);
            }
        }
        
        return ans;
    }
};